prog/mutation.go

```diff
@@ -184,6 +184,35 @@ func (ctx *mutator) mutateArg() bool {
 	return true
 }
 
+// Select a call based on the complexity of the arguments.
+func chooseCall(p *Prog, r *randGen) (*Call, bool) {
+	sum := 0
+	count := 0
+
+	var callsComplexity []int
+	for _, c := range p.Calls {
+		count = 0
+		ForeachArg(c, func(arg Arg, _ *ArgCtx) {
+			count++
```

Comment:
I think it's better to calculate priority for each syscall statically, that is, have it as part of Syscall struct here:
https://github.com/google/syzkaller/blob/8620c2c2581732b54045bd75ae1220982fbb650f/prog/types.go#L10
This should make mutation faster as we won't need to walk all syscalls and arguments each time. I think static priority should be good enough.
The calculation of call priorities should be in the prog package and use the getMutationPrio function. I think the priority should be something like a sum of priorities of all arguments. For unions we could take a maximum priority of all options. And for arrays maybe multiply element priority by 2 (because in some cases we won't know actual array size).
But the actual calculation and assignment should happen somewhere around this function:
https://github.com/google/syzkaller/blob/8620c2c2581732b54045bd75ae1220982fbb650f/pkg/compiler/gen.go#L48-L67
Then after 'make generate' you should see assigned priorities in the sys/linux/gen/amd64.go file.
And then mutation will use already precomputed values.